src/Microsoft.Health.Dicom.Functions/IndexMetricsCollection/IndexMetricsCollectionFunction.cs

```diff
@@ -0,0 +1,80 @@
+    private readonly bool _externalStoreEnabled;
+    private readonly bool _enableDataPartitions;
+
+    public IndexMetricsCollectionFunction(
+        IIndexDataStore indexDataStore,
+        IOptions<FeatureConfiguration> featureConfiguration,
+        IndexMetricsCollectionMeter meter)
+    {
+        EnsureArg.IsNotNull(featureConfiguration, nameof(featureConfiguration));
+        _indexDataStore = EnsureArg.IsNotNull(indexDataStore, nameof(indexDataStore));
+        _meter = EnsureArg.IsNotNull(meter, nameof(meter));
+        _externalStoreEnabled = featureConfiguration.Value.EnableExternalStore;
+        _enableDataPartitions = featureConfiguration.Value.EnableDataPartitions;
+    }
+
+
+    /// <summary>
+    /// Asynchronously collects index metrics.
+    /// </summary>
+    /// <param name="invocationTimer">The timer which tracks the invocation schedule.</param>
+    /// <param name="log">A diagnostic logger.</param>
+    /// <returns>A task that represents the asynchronous metrics collection operation.</returns>
+    [FunctionName(nameof(IndexMetricsCollectionFunction))]
+    public async Task Run(
+        [TimerTrigger(IndexMetricsCollectionOptions.Frequency)] TimerInfo invocationTimer,
+        ILogger log)
+    {
+        EnsureArg.IsNotNull(invocationTimer, nameof(invocationTimer));
+        EnsureArg.IsNotNull(log, nameof(log));
+
+        log.LogInformation("Collecting a daily summation starting. At: {Timestamp}", Clock.UtcNow);
+        if (invocationTimer.IsPastDue)
+        {
+            log.LogWarning("Current function invocation is running late.");
+        }
+
+        Stopwatch stopwatch = new Stopwatch();
+        stopwatch.Start();
+
+        IndexedFileProperties indexedFileProperties = await _indexDataStore.GetIndexedFilePropertiesAsync();
+
+        stopwatch.Stop();
+
+        log.LogInformation("Collecting a daily summation time taken: {ElapsedTime} ms with ExternalStoreEnabled: {ExternalStoreEnabled} and DataPartitionsEnabled: {PartitionsEnabled}", stopwatch.ElapsedMilliseconds, _externalStoreEnabled, _enableDataPartitions);
+
+        log.LogInformation("DICOM telemetry - total files indexed: {0} with ExternalStoreEnabled: {ExternalStoreEnabled} and DataPartitionsEnabled: {PartitionsEnabled}", indexedFileProperties.TotalIndexed, _externalStoreEnabled, _enableDataPartitions);
+
+        log.LogInformation("DICOM telemetry - total content length indexed: {0} with ExternalStoreEnabled: {ExternalStoreEnabled} and DataPartitionsEnabled: {PartitionsEnabled}", indexedFileProperties.TotalSum, _externalStoreEnabled, _enableDataPartitions);
+
+        log.LogInformation("Collecting a daily summation completed. with ExternalStoreEnabled: {ExternalStoreEnabled} and DataPartitionsEnabled: {PartitionsEnabled}", _externalStoreEnabled, _enableDataPartitions);
+
+        _meter.IndexMetricsCollectionsCompletedCounter.Add(1, IndexMetricsCollectionMeter.CreateTelemetryDimension(_externalStoreEnabled, _enableDataPartitions));
```

Comment:
Are other dimensions like the resource implied?

Reply:
I mean like as a metric by default this will have the aggregate as well as there two booleans. But on its own, we won't know what service had these metrics? Are those automatically applied for all metrics or do we need to include those here?